jib-core/src/main/java/com/google/cloud/tools/jib/frontend/MainClassFinder.java

```diff
@@ -51,21 +53,22 @@ public Class findClass(@Nullable String name) {
       try {
         // Name is only ever null when we call findClass manually, and not null otherwise. If null,
         // we should resolve the correct filename.
+        Path file = classFile;
         if (name != null) {
```

Comment:
I feel like this the null case should be in a separate method to avoid this null-check logic.